pkg/test/framework/components/istio/istio.go

```diff
@@ -35,7 +35,7 @@ type Instance interface {
 	// RemoteDiscoveryAddressFor returns the external address of the discovery server that controls
 	// the given cluster. This allows access to the discovery server from
 	// outside its cluster.
-	RemoteDiscoveryAddressFor(cluster resource.Cluster) (net.TCPAddr, error)
+	RemoteDiscoveryAddressFor(svcName string, cluster resource.Cluster) (net.TCPAddr, error)
```

Comment:
1. if we change this signature, the behavior of an empty service name should be well documented
2. If the istio component already has the `UsingRemoteKubeConfig` flag, couldn't we switch this to use the `istiodSvcName` in the implementation rather than requiring all callers? Most tests shouldn't need to know which service is doing discovery and can use this method to get that info from the environment.

Reply:
good idea! I am using `UsingRemoteKubeConfig` here to prevent modifying signature of this method.